### PR TITLE
homi: update docker compose to use daemon packages

### PR DESCRIPTION
**Pull request by jeonghonah**
## Proposed changes

- Update docker compose output of homi.
- After applying this patch, the docker compose uses daemon binaries (kcnd, kpnd, etc.) in packages.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: build/packaging/linux/bin/kscnd [hunk 1]
```diff
@@ -55,12 +55,6 @@ start() {
         mkdir -p $LOG_DIR
     fi
 
-    if [ -z $SCSIGNER_PASSWD_FILE ] || [ ! -f $SCSIGNER_PASSWD_FILE ]; then
```
Comment (jeonghonah) [inline build/packaging/linux/bin/kscnd]: This update is for 'istanbul' support in the config.

File: build/packaging/linux/conf/kscnd.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Configuration file for the kscnd
 
-SCSIGNER="0x0"
```
Comment (jeonghonah) [inline build/packaging/linux/conf/kscnd.conf]: It causes to fail empty scsigner.

File: cmd/homi/docker/service/validator.go [hunk 3]
```diff
@@ -123,51 +125,49 @@ var validatorTemplate = `{{ .Name }}:
...
-        --mainbridge \
-        --mainbridgeport 50505
+        echo 'SC_MAIN_BRIDGE=1' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'SC_MAIN_BRIDGE_PORT=50505' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'SC_MAIN_BRIDGE_INDEXING=1' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
 {{- end }}
-
+        echo 'ADDITIONAL="$$ADDITIONAL --debug --metrics --prometheus"' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        /klaytn-docker-pkg/bin/k{{ .NodeType }}d start
+        while true; do sleep 1; done
```
Comment (jeonghonah) [inline cmd/homi/docker/service/validator.go]: Trick is to keep containers running even node process is down.  Debugging purpose.

File: cmd/homi/docker/service/validator.go [hunk 4]
```diff
@@ -123,51 +125,49 @@ var validatorTemplate = `{{ .Name }}:
...
-        --syncmode "full" \
+        echo "# docker-compose" >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'NETWORK=""' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'DATA_DIR="/klaytn"' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'LOG_DIR="$$DATA_DIR/log"' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'RPC_ENABLE=1' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'RPC_API="db,klay,net,web3,miner,personal,txpool,debug,admin,istanbul,mainbridge,subbridge"' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'NETWORK_ID="{{ .NetworkId }}"' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'NO_DISCOVER=1' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
+        echo 'SYNCMODE="full"' >> /klaytn-docker-pkg/conf/k{{ .NodeType }}d.conf
```
Comment (KimKyungup) [inline cmd/homi/docker/service/validator.go]:
```suggestion
```
We can skip this by default value. :)

File: cmd/homi/docker/service/validator.go [hunk 5]
```diff
@@ -123,51 +123,48 @@ var validatorTemplate = `{{ .Name }}:
...
         echo "SuperSecret1231" > /klaytn/password.txt
 {{- end}}
-        k{{ .NodeType }} \
-        --identity "{{ .Name }}" \
-        --rpc \
-        --rpcaddr "0.0.0.0" \
-        --rpcport "8551" \
-        --rpccorsdomain "*" \
-        --datadir "/klaytn" \
-        --port "32323" \
```
Comment (laggu) [inline cmd/homi/docker/service/validator.go]: It seems the port is only used on scn
Is the port not required by other nodes??

Comment (jeonghonah) [reply cmd/homi/docker/service/validator.go]: Yes, default value is used.

**Pull request merged**